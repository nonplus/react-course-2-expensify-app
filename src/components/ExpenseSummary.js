import React from "react";
import { connect } from "react-redux";
import numeral from "numeral";
import { selectExpenses } from "../selectors/expenses";
import { selectExpensesTotal } from "../selectors/expenses-total";

export const ExpenseSummary = ({ expenseCount, expensesTotal }) => {
  const expenseWord = expenseCount === 1 ? "expense" : "expenses";
  const formatedCount = numeral(expenseCount).format("0,0");
  const formatedTotal = numeral(expensesTotal / 100).format("$0,0.[99]");
  return (
    <div>
      <h1>
        You're viewing {formatedCount} {expenseWord} totaling {formatedTotal}
      </h1>
    </div>
  );
};

const mapStateToProps = state => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses)
  };
};

export default connect(mapStateToProps)(ExpenseSummary);
