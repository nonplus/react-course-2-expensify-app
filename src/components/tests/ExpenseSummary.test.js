import React from "react";
import { shallow } from "enzyme";
import { ExpenseSummary } from "../ExpenseSummary";

describe("<ExpenseSummary />", () => {
  it("should render summary for 0 expenses", () => {
    const wrapper = shallow(
      <ExpenseSummary expenseCount={0} expensesTotal={0} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("should render summary for 1 expense", () => {
    const wrapper = shallow(
      <ExpenseSummary expenseCount={1} expensesTotal={123.45} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("should render summary for 2 expenses", () => {
    const wrapper = shallow(
      <ExpenseSummary expenseCount={2} expensesTotal={123.45} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("should render summary for thousands of expenses", () => {
    const wrapper = shallow(
      <ExpenseSummary expenseCount={12345} expensesTotal={123456789} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
