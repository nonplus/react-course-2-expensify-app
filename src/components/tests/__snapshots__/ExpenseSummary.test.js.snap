// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExpenseSummary /> should render summary for 0 expenses 1`] = `
<div>
  <h1>
    You're viewing 
    0
     
    expenses
     totaling 
    $0
  </h1>
</div>
`;

exports[`<ExpenseSummary /> should render summary for 1 expense 1`] = `
<div>
  <h1>
    You're viewing 
    1
     
    expense
     totaling 
    $1.23
  </h1>
</div>
`;

exports[`<ExpenseSummary /> should render summary for 2 expenses 1`] = `
<div>
  <h1>
    You're viewing 
    2
     
    expenses
     totaling 
    $1.23
  </h1>
</div>
`;

exports[`<ExpenseSummary /> should render summary for thousands of expenses 1`] = `
<div>
  <h1>
    You're viewing 
    12,345
     
    expenses
     totaling 
    $1,234,567.89
  </h1>
</div>
`;
